Improvement: Improving the way to extract arguments while raising errors

shubhamzanwar opened the pull request:
![image](https://user-images.githubusercontent.com/15626155/54882680-7bb9cd00-4e82-11e9-859f-4b40a51e33be.png)

testcase summary for reference

nelsonic commented on lib/index.js:
```diff
@@ -104,23 +104,18 @@ internals.implementation = function(server, options) {
+    const _errorType = errorContext.errorType;
+    const _message = errorContext.message;
+    const _scheme = errorContext.scheme;
+    const _attributes = errorContext.attributes;
```
Why define four `const`? if we're trying to improve/optimise this 
could we simply inline them in the `return` statement and avoid using up memory?

nelsonic commented on lib/index.js:
```diff
@@ -104,23 +104,18 @@ internals.implementation = function(server, options) {
+    // the errorFunc must return an object with keys:
+    // errorType and message, we need not worry about
+    // errorContext being undefined
+    const _errorType = errorContext.errorType;
```
only the `_errorType` is required, the remaining 3 `const`s can be in-lined. 👍

nelsonic commented on lib/index.js:
```diff
@@ -104,23 +104,18 @@ internals.implementation = function(server, options) {
-    let _attributes = attributes; // if you know a better way please PR!
 
-    if (options.errorFunc && internals.isFunction(options.errorFunc)) {
+    if (internals.isFunction(options.errorFunc)) {
```
The idea behind having this check is that we test for presence of the key before testing for type.
But you're _right_, the `internals.isFunction` does this check already, so OK to remove. 👍

shubhamzanwar commented:
@nelsonic I've updated the PR as per your comments.
I've removed all four declared constants in order to maintain consistency. I'm using them directly while calling the Boom error.

Also, the documentation mentions that both errorType and message are required for the options.errorFunc function:
![image](https://user-images.githubusercontent.com/15626155/54884337-a95c4180-4e95-11e9-8b7c-954091d77a3a.png)

Is this something we should update in the docs?
